encord/utilities/benchmark_utilities.py

```diff
@@ -0,0 +1,168 @@
+import dataclasses
+
+from shapely.geometry import Point, Polygon, box
+
+from encord.orm.label_row import LabelRow
+
+
+@dataclasses.dataclass
+class LabelAnnotationMetrics:
+    precision: float
+    recall: float
+
+
+# ---------------------------------------------------------
+#                   Helper functions
+# ---------------------------------------------------------
+def add_frame_consensus_score(base_frame, comparing_frames, label_types, threshold, consensus_score):
+    """
+    Update consensus score when comparing base_frame's objects with consensus objects obtained from comparing_frames.
+    Two objects' instances are considered the same object if their Jaccard similarity coefficient is greater or equal
+    than the threshold; otherwise, they represent distinct objects.
+    """
+    consensus_objects = find_consensus_objects(comparing_frames, label_types, threshold)
+    # Score each base_frame's object as true positive (TP) or false positive (FP) when compared to consensus objects
+    for base_object in base_frame["objects"]:
+        feature_hash = base_object["featureHash"]
+        if feature_hash not in label_types:
+            continue
+        # Compare base_object with consensus objects of the same type to search for base_object's most similar object
+        max_similarity = (-1, -1)  # (best_jaccard_similarity_coefficient, index_of_selected_consensus_object)
+        consensus_objects.setdefault(feature_hash, dict())
+        for comparing_index, comparing_object in consensus_objects[feature_hash].items():
+            cur_similarity = (calculate_jaccard_similarity(base_object, comparing_object), comparing_index)
+            max_similarity = max(max_similarity, cur_similarity)
+        score = consensus_score.setdefault(feature_hash, {"TP": 0, "FP": 0, "FN": 0})
+        if max_similarity[0] >= threshold:
+            score["TP"] += 1
+            consensus_objects[feature_hash].pop(max_similarity[1])
+        else:
+            score["FP"] += 1
+    # Score each unmatched consensus object as false negative (FN)
+    for feature_hash, objects in consensus_objects.items():
+        score = consensus_score.setdefault(feature_hash, {"TP": 0, "FP": 0, "FN": 0})
+        score["FN"] += len(objects)
+
+
+def find_consensus_objects(comparing_frames, label_types, threshold):
```

Comment:
Did you drop the `__` because someone might use this function?

Reply:
No, I drop the `__` because some errors I got when the main method was transferred to `client.py`. In fact, the helper functions that someone might use and shouldn't have `__` are `calculate_jaccard_similarity` and `transform_obj_to_polygon`. I am going to prefix `__` to `find_consensus_objects` and `add_frame_consensus_score` functions.

Reply:
Fixed. Now we have `__add_frame_consensus_score`, `__find_consensus_objects`, `calculate_jaccard_similarity` and `transform_object_dict_to_polygon` (renamed to better comprehension)